File: web-app/data-core/services/kafka-service.ts
import { Kafka, RecordMetadata } from "kafkajs";
const kafka = new Kafka({
    clientId: process.env.KAFKA_CLIENT_ID!,
    brokers: [process.env.KAFKA_BROKER!]
})
export class KafkaService {
    static async sendMessageToKafka(topic: string, message: string): Promise<boolean> {
        try {
            const producer = kafka.producer()
            await producer.connect()
            const response: RecordMetadata[] = await producer.send({
                topic: topic,
                messages: [{ value: message }]
            })
            console.log('Message sent to kafka', response)
            await producer.disconnect();
            return true;
        } catch (error) {
            throw error;
        }
    }
}

File: video-processing-service/app/services/kafka_service.py
from kafka import KafkaProducer
import json
from dotenv import load_dotenv
import os

load_dotenv()

# Kafka configuration
KAFKA_BROKER = os.getenv("KAFKA_BROKER")


# Create a Kafka producer
producer = KafkaProducer(
    bootstrap_servers=KAFKA_BROKER,
    value_serializer=lambda v: json.dumps(v).encode("utf-8"),
)


class KafkaService:
    @staticmethod
    async def send_message_to_kafka(data):
        print("Sending message to Kafka", data)
        result = producer.send("process-video-update", value=json.loads(data))
        producer.flush()
        print("this is the result of sending to kafka", result)
        print("Message sent to Kafka")
